fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -897,6 +903,29 @@ ACTOR Future<Void> getResolution(CommitBatchContext* self) {
 	self->transactionResolverMap.swap(requests.transactionResolverMap);
 	// Used to report conflicting keys
 	self->txReadConflictRangeIndexMap.swap(requests.txReadConflictRangeIndexMap);
+
+	// Fetch cipher keys if needed.
+	state Future<std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>>> getCipherKeys;
+	if (SERVER_KNOBS->ENABLE_TLOG_ENCRYPTION) {
+		std::unordered_map<EncryptCipherDomainId, EncryptCipherDomainName> encryptDomains = {
+			{ SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID, FDB_DEFAULT_ENCRYPT_DOMAIN_NAME },
+			{ ENCRYPT_HEADER_DOMAIN_ID, FDB_DEFAULT_ENCRYPT_DOMAIN_NAME }
+		};
+		for (int t = 0; t < trs.size(); t++) {
+			int64_t tenantId = trs[t].tenantInfo.tenantId;
+			Optional<TenantName> tenantName = trs[t].tenantInfo.name;
+			if (tenantId != TenantInfo::INVALID_TENANT) {
+				if (!tenantName.present()) {
```

Comment:
assert?

Reply:
Instead a more explicit error is thrown. Is it better?

Reply:
ASSERT is more like operation is `unacceptable`; error thrown would qualify in acceptable situations. `TenantName` is a _must-have_, I think ASSERT would be better. Thoughts?